velox/exec/GroupingSet.cpp

```diff
@@ -412,4 +444,245 @@ const HashLookup& GroupingSet::hashLookup() const {
   return *lookup_;
 }
 
+namespace {
+bool maybeReserve(int64_t increment, memory::MemoryUsageTracker& tracker) {
```

Comment:
Any particular reason to have this logic inside of the aggregation operator and not as a method on MemoryUsageTracker?